pkg/ocm/ocm.go

```diff
@@ -418,3 +414,99 @@ func HashIdentity(id ocmmetav1.Identity) (string, error) {
 
 	return fmt.Sprintf("sha-%d", hash), nil
 }
+
+type componentBundle struct {
+	versioned ocm.ComponentVersionAccess
+	list      ocm.ComponentAccess
+}
+
+func (c *componentBundle) Close() error {
+	if c.versioned != nil {
+		return c.versioned.Close()
+	}
+
+	return c.list.Close()
+}
+
+type lookupComponentOption struct {
```

Comment:
if this is to be internal only maybe we don't need the functional option here. We can refactor those without breaking consumers code.

Reply:
The functional options are there because sometimes it's called with a version and sometimes not. Sometimes there is a name sometimes not. And I think passing in `""` just looked ugly. :D So that's why I choose these options. :) WDYT?

Reply:
functional options are used to avoid breaking api contract. Here you clearly don't have that use case. You can just pass in the struct like:

```go
func (c *Client) lookupComponent(ctx context.Context, octx ocm.Context, obj *v1alpha1.ComponentVersion, opts lookupComponentOption) (*componentBundle, error) {
```

this way you don't need the extra `funcs`

Reply:
Functional options are not just for breaking apis. They are also there for slick API design with variable settings and in the mind of "define it only when needed". With the struct, you always have to pass in the struct. Here for example:

```go
func (c *Client) listComponentVersions(logger logr.Logger, octx ocm.Context, obj *v1alpha1.ComponentVersion) ([]Version, error) {
	cv, err := c.lookupComponent(context.Background(), octx, obj)
	if err != nil {
		return nil, fmt.Errorf("failed to lookup componint: %w", err)
	}
```

It would mean I would have to pass in an empty struct without reason like `lookupComponentOption{}`. Which is ugly IMHO. Since the name doesn't need to be overwritten and we don't require a version.

Reply:
I understand your point of view. It would save you 2 `withFunc` functions though. Your choice.

Reply:
I'll think about it. :D I'll see how I like it. :)

Reply:
Alright, I re-wrote it. :) It does look better, and less code. :)